### PR TITLE
Release/3.0.1

### PR DESCRIPTION
**Pull request by rpdome**

### CODE REVIEW

File: userappwithbroker/build.gradle [hunk 1]
```diff
@@ -70,7 +70,7 @@ dependencies {
     implementation "com.google.android.material:material:$rootProject.ext.materialVersion"
 
     localImplementation(project(':adal'))
-    distImplementation("com.microsoft.aad:adal:1.16.3-hf2") {
+    distImplementation("com.microsoft.aad:adal:3.0.1-RC1") {
```
Comment (iambmelt) [inline userappwithbroker/build.gradle:73]: Nit: should this be updated to the prod build?

**Comment by shahzaibj** [comment]
Do we need to update the changelog @rpdome ? I'm not seeing anything

**Comment by rpdome** [comment]
Changelog is still TBD (Waiting for a signal from MAM team).

**Comment by iambmelt** [comment]
@rpdome jfyi it looks like CI is failing

File: adal/build.gradle [hunk 2]
```diff
@@ -132,16 +132,17 @@ dependencies {
...
     localApi(project(":common")) {
         transitive = false
     }
 
-    snapshotApi(group: 'com.microsoft.identity', name: 'common', version: '2.0.3', changing: true)
+    snapshotApi(group: 'com.microsoft.identity', name: 'common', version: '2.0.4', changing: true)
 
     // 'dist' flavor dependencies
-    distApi("com.microsoft.identity:common:2.0.3") {
+    distApi("com.microsoft.identity:common:2.0.5") {
```
Comment (iambmelt) [inline adal/build.gradle:145]: Snapshot & dist are on different versions. Intentional?

Comment (rpdome) [reply adal/build.gradle:145]: Nope. fixed.

File: changelog.txt [hunk 3]
```diff
@@ -1,3 +1,14 @@
+Version 3.0.1
+-------------
+- Adding dual screen support for Surface Duo.
+  * Please exclude classes under "com.microsoft.device.display" in your proguard file.
+  * For apps that would like to include their own copy of Surface Duo SDK, please exclude the SDK inside ADAL as follows.
```
Comment (iambmelt) [inline changelog.txt:5]: I think this needs to be front & center in the readme -- we did the same thing for Gson here:
https://github.com/AzureAD/azure-activedirectory-library-for-android#option-2-binaries-via-gradle

File: README.md [hunk 4]
```diff
@@ -404,10 +404,7 @@ If you are using ProGuard, we suggest disabling shrinking and obfuscation of ADA
 To disable obfuscation of ADAL classes:
 
 ```
--keep class com.microsoft.aad.adal.** {
-  *;
-}
--keep class com.microsoft.identity.common.** {
+-keep class com.microsoft.** {
```
Comment (iambmelt) [inline README.md:407]: There's some more lines in this file I would expect to see changed
- Current version designator
- Minimum recommended version (probably 1.16.1 at this point)
- [This](https://github.com/AzureAD/azure-activedirectory-library-for-android#option-2-binaries-via-gradle)

Comment (rpdome) [reply README.md:407]: Should update minimum version to 1.16.3-hf2?

Comment (iambmelt) [reply README.md:407]: Can't remember what the last crit fix for us was, I'll defer to your better judgement 😄

File: adal/src/main/java/com/microsoft/aad/adal/AuthenticationActivity.java [hunk 5]
```diff
@@ -122,15 +121,15 @@
...
+public class AuthenticationActivity extends DualScreenActivity {
 
     static final int BACK_PRESSED_CANCEL_DIALOG_STEPS = -2;
 
     private static final String TAG = "AuthenticationActivity";
     private boolean mRegisterReceiver = false;
     private WebView mWebView;
     private String mStartUrl;
-    private ProgressDialog mSpinner;
+    private ProgressBar mSpinner;
```
Comment (iambmelt) [inline adal/src/main/java/com/microsoft/aad/adal/AuthenticationActivity.java:132]: Let's rename this property, since it's not a `Spinner` anymore

Comment (rpdome) [reply adal/src/main/java/com/microsoft/aad/adal/AuthenticationActivity.java:132]: It's still a spinner actually. Just not in a dialog box.

Comment (iambmelt) [reply adal/src/main/java/com/microsoft/aad/adal/AuthenticationActivity.java:132]: @rpdome [Spinner](https://developer.android.com/guide/topics/ui/controls/spinner)'s a chooser box, not a progress indicator

Comment (iambmelt) [reply adal/src/main/java/com/microsoft/aad/adal/AuthenticationActivity.java:132]: I guess the name was always wrong? 🤦‍♂

Comment (rpdome) [reply adal/src/main/java/com/microsoft/aad/adal/AuthenticationActivity.java:132]: ahhh.... what I meant is that it's a spinner as in... it's a spinning progress bar 🤣

Comment (rpdome) [reply adal/src/main/java/com/microsoft/aad/adal/AuthenticationActivity.java:132]: but yeah this has always been the name.

**Pull request merged**